src/options/client_options.ts

```diff
@@ -0,0 +1,40 @@
+const url = require('url');
+const qs = require('querystring');
+import type { UriOptions, ClientOptions } from './types';
+import { CoerceCustom } from './coerce_custom';
+
+export class MongoClientOptions {
+  static parseConnectionString(connectionString: string) {
```

Comment:
This is not finalized, needs errors and I will go over it again and add test specific to this function.